provides.py

```diff
@@ -30,10 +30,11 @@ def departed(self):
         conv = self.conversation()
         conv.remove_state('{relation_name}.joined')
 
-    def set_ready(self, port, controllers):
+    def set_ready(self, port, api_path, controllers):
```

Comment:
It occurs to me that we should get in the habit of only setting one thing per method call, or use optional keyword params, to avoid breaking backwards compatibility.  No problem with this change, just something to think about for the future.